packages/@ember/runloop/index.ts

```diff
@@ -99,8 +128,20 @@ export const _backburner = new Backburner(_queues, {
   @return {Object} return value from invoking the passed function.
   @public
 */
-export function run() {
-  return _backburner.run(...arguments);
+export function run<F extends (...args: any[]) => any>(method: F, ...args: Parameters<F>): void;
+export function run<T, F extends (this: T, ...args: any[]) => any>(
```

Comment:
We should check how this works compared to what I implemented on DT for Ember v4 types. The version on DT has some limitations, so this may be preferable. (Or you may have stolen it from there! I don’t remember exactly what I wrote. :joy:)

Reply:
I did my own research 😝